website/docs/r/kubernetes_cluster.html.markdown

```diff
@@ -448,6 +450,19 @@ An `identity` block supports the following:
 
 ---
 
+A `key_vault_kms` block supports the following:
+
+* `enabled` - (Optional) Whether to enable Azure Key Vault key management service. The default is `false`.
+
+* `key_id` - (Optional) Identifier of Azure Key Vault key. See [key identifier format](https://learn.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#vault-name-and-object-name) for more details. When Azure Key Vault key management service is enabled, this field is required and must be a valid key identifier. When `enabled` is `false`, leave the field empty.
```

Comment:
I would recommend adding the relevant check in aks resource's `expandKubernetesClusterAzureKeyVaultKms` function.

Reply:
Not required. It is catched by the Schema:
```golang
						"enabled": {
							Type:     pluginsdk.TypeBool,
							Required: true, // <-- here
						},
						"key_id": {
							Type:         pluginsdk.TypeString,
							Required:     true, // <-- and here
							ValidateFunc: keyVaultValidate.NestedItemId,
						},
```